vcd/resource_vcd_vapp.go

```diff
@@ -493,32 +473,50 @@ func resourceVcdVAppDelete(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("error finding vapp: %s", err)
 	}
 
-	_ = retryCall(vcdClient.MaxRetryTimeout, func() *resource.RetryError {
-		task, err := vapp.Undeploy()
-		// Very often the vApp is powered off at this point and Undeploy() would wait for as long as
-		// vcdClient.MaxRetryTimeout before going to Delete phase because of the following error:
-		// "The requested operation could not be executed since vApp vApp_name is not running"
-		// So, if the error matches we just quit retrying and fast forward to vapp.Delete()
-		var reErr = regexp.MustCompile(`.*The requested operation could not be executed since vApp.*is not running.*`)
-		if err != nil && reErr.MatchString(err.Error()) {
-			return &resource.RetryError{Err: err, Retryable: false}
-		}
+	// to avoid network destroy issues - detach networks from VAPP
+	task, err := vapp.RemoveAllNetworks()
+	if err != nil {
+		return fmt.Errorf("error with Networking change: %#v", err)
+	}
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return fmt.Errorf("error changing network: %#v", err)
+	}
 
-		if err != nil {
-			return resource.RetryableError(fmt.Errorf("error undeploying: %#v", err))
-		}
+	err = tryUndeploy(vapp)
+	if err != nil {
+		return err
+	}
+
+	task, err = vapp.Delete()
+	if err != nil {
+		return fmt.Errorf("error deleting: %#v", err)
+	}
 
-		return resource.RetryableError(task.WaitTaskCompletion())
-	})
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return fmt.Errorf("error with deleting VAPP task: %#v", err)
```

Comment:
Nit, but please use "vApp" instead of "VAPP" in the messages and comments, there are more occurrences like these in this PR.

Reply:
fixed